### ISSUE TITLE
CloseButton component is missing accessible label

### ISSUE THREAD

**Issue by khiga8** [primer/view_components#460]
[ARIA14: Using aria-label to provide an invisible label where a visible label cannot be used](https://www.w3.org/TR/WCAG20-TECHS/ARIA14.html#:~:text=The%20'close'%20element%20is%20implemented,accessible%20name%20to%20the%20button.)

The `CloseButton` component is missing an accessible label so the screen reader will simply announce "button". If we are not providing an aria-label ourselves, we should update the documentation to recommend use of label.
